### PR TITLE
check for OOB read in getSubmission/getComment

### PR DESCRIPTION
**Pull request by ghost**
Hey,

I noticed that mira panics when trying to `r.Submission("t3_asdf12").Info()` when `t3_asdf12` is an invalid/unknown id, since the json returned is empty. I added a simple check that throws an error that can easily be caught in the app using mira.

Greetings,
u/try_to_guess_my_psn

### CODE REVIEW

File: reddit.go [hunk 1]
```diff
@@ -143,6 +143,9 @@ func (c *Reddit) getSubmission(id string) (models.PostListingChild, error) {
 	})
 	ret := &models.PostListing{}
 	json.Unmarshal(ans, ret)
+	if len(ret.GetChildren()) < 1 {
+		return "", fmt.Errorf("id not found")
```
Comment (thecsw) [inline reddit.go]: Build failed: https://travis-ci.com/thecsw/mira/builds/130496105

```
github.com/thecsw/mira
235# github.com/thecsw/mira
236./reddit.go:147:10: cannot use "" (type string) as type models.PostListingChild in return argument
237./reddit.go:160:10: cannot use "" (type string) as type models.Comment in return argument
```

Change the empty string to `nil`

**Review by thecsw — CHANGES_REQUESTED** [review]
Change the empty string to `nil` and we are good to go!

**Comment by ghost** [comment]
oooh sorry. This is what happens when you edit online ;)

it returns an empty value now. Travis runs through, so it builds now :)

**Review by thecsw — APPROVED** [review]
Good stuff! Approved. Thank you for spotting the bug

**Pull request merged**